src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -139,6 +139,19 @@ class TypeComparer(initctx: Context) extends DotClass with ConstraintHandling wi
   }
 
   private def firstTry(tp1: Type, tp2: Type): Boolean = tp2 match {
+    // FIXME: This is needed because sometimes (underlying1 ne underlying2)
```

Comment:
have you tried `widenDealias`? can you show a test case when `underlying1.info.widenDealias ne underlying1.info.widenDealias`?